### PR TITLE
Add iseeu options to plotDots

### PR DESCRIPTION
**Pull request by alanocallaghan**

### CODE REVIEW

**Comment by LTLA** [comment]
Just nag me when you think it's done and I'll have a look.

**Comment by alanocallaghan** [comment]
I think I got everything but if I missed any spots just holler

File: R/plotDots.R [hunk 1]
```diff
@@ -101,6 +111,18 @@ plotDots <- function(object, features, group = NULL, block=NULL, exprs_values =
         group.names <- colnames(ave)
     }
 
+    if (center) {
+        ave <- ave - rowMeans(ave)
+    }
+    if (scale) {
+        ave <- ave / apply(ave, 1, function(row) {
+            sqrt(sum(row^2) / (length(row) - 1))
+        })
```
Comment (LTLA) [inline R/plotDots.R]: Probably could do this more elegantly with just:

```r
ave <- ave/sqrt(rowSums(ave^2) / (ncol(ave) - 1))
```

File: R/plotDots.R [hunk 2]
```diff
@@ -101,6 +111,18 @@ plotDots <- function(object, features, group = NULL, block=NULL, exprs_values =
...
+    }
+    if (!is.null(max_ave)) {
+        ave <- pmin(max_ave, ave)
```
Comment (LTLA) [inline R/plotDots.R]: We could make this more consistent with how `plotHeatmap()` does it, using a `zlim` argument to control the color scale in a manner dependent on `symmetric=`. (You could probably even spin this out into a separate utility that is used across all these related functions.) 

Though it does raise the question of whether `center!=symmetric` _ever_ makes any sense for a color scale.

File: R/plotDots.R [hunk 3]
```diff
@@ -111,8 +133,17 @@ plotDots <- function(object, features, group = NULL, block=NULL, exprs_values =
     if (!is.null(max_detected)) {
         evals_long$NumDetected <- pmin(max_detected, evals_long$NumDetected)
     }
-    if (!is.null(max_ave)) {
-        evals_long$Average <- pmin(max_ave, evals_long$Average)
+    if (scale) {
```
Comment (LTLA) [inline R/plotDots.R]: Did you mean this to be `center`? Arguably the most consistent approach would be to have a `symmetric=center` default in the argument list... assuming that a separate `symmetric` argument even makes sense, see comments above.

I also wonder whether we should have more consistent default color schemes across all these related plots (`plotHeatmap()`, this, and `plotGroupedHeatmap()`). You could imagine creating another utility function that chooses the default color scheme depending on whether `center=TRUE`; if yes, we use whatever palette **pheatmap** uses, which looks nice enough, while for `FALSE`, we can default to viridis.

**Comment by LTLA** [comment]
What do you think about rolling out all these related options into a separate internal function, e.g., that handles centering, scaling, enforcement of `zlim` and selection of an appropriate default color scale?

**Comment by alanocallaghan** [comment]
Makes sense I just wanted to square off the interface for plotDots before refactoring

**Comment by LTLA** [comment]
Very nice. Will play around with it tonight.

**Comment by LTLA** [comment]
Alright, cleaned it up a little on my end. Put back the discarded options as deprecated things so people who were using them in the current release get a soft landing. Also made use of `@inheritParams` to share parameter definitions across these three functions.

**Pull request merged**